### PR TITLE
Add error code for update distribution error

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR adds a minimal change to return a error code when updating distribution for dataset with existing responses.

**Type of change**

- Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by frascuchon** [comment]
@jfcalvo I saw some [TODO comments](https://github.com/argilla-io/argilla/pull/5258/files#diff-86e2a154e3c4de51f12c5772405f5424211d363d4628ee08b134caacc906f0e9R55) about the error handling that should be tackled for the v2.0 but I don't understand.

File: argilla-server/src/argilla_server/errors/future/base_errors.py [hunk 1]
```diff
@@ -55,6 +57,11 @@ class MissingVectorError(UnprocessableEntityError):
     pass
 
 
+class UpdateDistributionWithExistingResponsesError(UnprocessableEntityError):
+    def __init__(self, message):
+        super().__init__(message, code=UPDATE_DISTRIBUTION_WITH_EXISTING_RESPONSES_ERROR_CODE)
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/errors/future/base_errors.py:62]: Why we need this error to have an specific code and being managed in a special way?

Comment (frascuchon) [reply argilla-server/src/argilla_server/errors/future/base_errors.py:62]: As we do for missing vectors, the frontend need to handle this specific error when trying to update distribution from the UI

File: argilla-server/src/argilla_server/api/errors/v1/exception_handlers.py [hunk 2]
```diff
@@ -67,3 +67,9 @@ async def missing_vector_error_exception_handler(request, exc):
             status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
             content={"code": exc.code, "message": exc.message},
         )
+
+    @app.exception_handler(errors.UpdateDistributionWithExistingResponsesError)
+    async def update_distribution_with_existing_responses(request, exc):
+        return JSONResponse(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, content={"code": exc.code, "message": exc.message}
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/api/errors/v1/exception_handlers.py]:
```suggestion
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, 
            content={"code": exc.code, "message": exc.message},
```

**Review by jfcalvo — APPROVED** [review]

**Pull request merged**